src/openApi/apiResponseModel.ts

```diff
@@ -3,11 +3,16 @@ export class ApiResponse<T = any> {
 
   public readonly statusCode: number;
 
-  public readonly headers?: Record<string, string>;
+  public readonly headers: Record<string, string>;
 
   constructor(statusCode: number, body: T, headers?: Record<string, string>) {
     this.body = body;
     this.statusCode = statusCode;
-    this.headers = headers;
+    this.headers = headers ?? {};
+  }
+
+  public withCorrelationId(correlationId: string) {
+    this.headers.correlationId = correlationId;
```

Comment:
This is probably a wrong header name, right?

Reply:
Good catch, fixed!